utils/aws-operations.js

```diff
@@ -0,0 +1,93 @@
+const AWS = require('aws-sdk');
+const config = require('config');
+
+const {
+  accessKeyId,
+  secretAccessKey,
+  region,
+  endpoint,
+  s3ForcePathStyle,
+} = config.get('aws');
+
+const s3 = new AWS.S3({
+  apiVersion: '2006-03-01',
+  accessKeyId,
+  secretAccessKey,
+  region,
+  endpoint,
+  s3ForcePathStyle,
+});
+this.bucket = null;
+
+/**
+ * @summary Set the bucket to be used for subsequent function calls.
+ * @function
+ */
+const setBucket = (bucket) => {
+  this.bucket = bucket;
+};
+
+/**
+ * @summary Checks if a bucket exists
+ * @function
+ * @param bucket The bucket to be checked
+ * @returns {Promise} Promise object represents a boolean indicating if the bucket exists or not
+ */
+const bucketExists = (bucket = this.bucket) => new Promise((resolve, reject) => {
+  const params = { Bucket: bucket };
+  // TODO: use valid credentials
+  s3.headBucket(params).promise().then(() => {
+    resolve(true);
+  }).catch((err) => {
+    if (err.code === 'NotFound') {
+      resolve(false);
+    } else {
+      reject(err);
+    }
+  });
+});
+
+/**
+ * @summary Checks if an object exists in a bucket
+ * @function
+ * @param key The key of the object to be checked
+ * @param bucket The bucket where the key will be searched
+ * @returns {Promise} Promise object represents a boolean indicating if the key exists or not
+ */
+const objectExists = (key, bucket = this.bucket) => new Promise((resolve, reject) => {
+  const params = { Bucket: bucket, Key: key };
+  // TODO: use valid credentials
+  s3.headObject(params).promise().then(() => {
+    resolve(true);
+  }).catch((err) => {
+    if (err.code === 'NotFound') {
+      resolve(false);
+    } else {
+      reject(err);
+    }
+  });
+});
+
+/**
+ * @summary Gets an object from a bucket
+ * @function
+ * @param key The key of the object
+ * @param bucket The bucket where the object exists
+ * @returns {Promise} Promise object representing the object response
+ */
+const getObject = (key, bucket = this.bucket) => new Promise((resolve, reject) => {
+  const params = { Bucket: bucket, Key: key };
+  // TODO: use valid credentials
+  s3.getObject(params).promise().then((data) => {
+    resolve(data);
+  }).catch((err) => {
+    reject(err);
+  });
+});
+
+module.exports = {
+  setBucket,
+  bucketExists,
+  objectExists,
+  getObject,
+};
```

Comment:
Thinking about it more, I feel like we should place this file to `/api/v1/db/aws` like the structure [here](https://github.com/osu-mist/new-students-api/tree/feature/CO-1340-remake-students-api/api/v1/db/oracledb). Mainly because I feel like the functionality is pretty similar than the `connection.js` in `oracledb` folder. Same thing, if we still have `utils/fs-operations.js` in the skeleton, then it should probably go to `/api/v1/db/json`. What do you think?

Reply:
I think this makes sense. Some of the original intent of `fs-operations` was to have some generic file operation functions that wrapped `fs` methods, partly for testing. Most of the functions in there now though are specific to JSON files or to the notes api local JSON database itself. Because of that, I don't think it needs to stay in `utils/`. `aws-operations` only has functions specific to the AWS implementation so I think it would be fine to move it also.